src/extensions/tracing-headers.ts

```diff
@@ -0,0 +1,104 @@
+import { PostHog } from '../posthog-core'
+import { patch } from './replay/rrweb-plugins/patch'
+import { window } from '../utils/globals'
+import { logger } from '../utils/logger'
+import { SessionIdManager } from '../sessionid'
+
+export class TracingHeaders {
+    private _restoreXHRPatch: (() => void) | undefined = undefined
+    private _restoreFetchPatch: (() => void) | undefined = undefined
+
+    constructor(private readonly instance: PostHog) {}
+
+    public startIfEnabledOrStop() {
+        if (this.instance.config.__add_tracing_headers && this._canPatch()) {
+            // we can assert this is present because we've checked previously
+            this._patchXHR(this.instance.sessionManager!)
+            this._patchFetch(this.instance.sessionManager!)
+        } else {
+            this._restoreXHRPatch?.()
+            this._restoreFetchPatch?.()
+            this._restoreXHRPatch = undefined
+            this._restoreFetchPatch = undefined
```

Comment:
The only thing that isn't clear to me is why we call and then reset these two functions, specifically in the case that `_canPatch` fails because we have already patched XHR and Fetch

Reply:
slightly copy pasta 🙈 

i think it's so we don't try to call them twice